### PR TITLE
Check the map for all GameConstants MAP CONSTANTS

### PR DESCRIPTION
**Pull request by stefangimmillaro**
Since the last one looked good I figured I would add the checks for the other constraints specified here https://github.com/battlecode/battlecode22/blob/main/engine/src/main/battlecode/common/GameConstants.java#L14

This one is a slightly more involved so please look carefully as I did this quickly (the same amount of time it took me to drink my morning 🍵) without specs.

Cheers and again thank you.

### CODE REVIEW

**Comment by stefangimmillaro** [comment]
and thank you @wflms20110333 for the previous review!

File: engine/src/main/battlecode/server/Server.java [hunk 1]
```diff
@@ -194,7 +196,44 @@ public void run() {
...
+                throw new RuntimeException("RUBBLE EXCEEDS GameConstants.MAX_RUBBLE");
+            }
+        }
+        
+        // Check starting Archons
+        int archonCount = 0;
+        for (RobotInfo robotInfo : liveMap.getInitialBodies()) {
+            if (robotInfo.type == RobotType.ARCHON) archonCount++;
+        }
+        if (archonCount < GameConstants.MIN_STARTING_ARCHONS*2) {
```
Comment (wflms20110333) [inline engine/src/main/battlecode/server/Server.java:230]: style comment: can you put a space around the multiplication operator lol

File: engine/src/main/battlecode/server/Server.java [hunk 2]
```diff
@@ -194,7 +196,44 @@ public void run() {
...
+            throw new RuntimeException("RUBBLE BENEATH GameConstants.MIN_STARTING_ARCHONS");
+        }
+        if (archonCount > GameConstants.MAX_STARTING_ARCHONS*2) {
```
Comment (wflms20110333) [inline engine/src/main/battlecode/server/Server.java:233]: ditto

**Review by wflms20110333 — CHANGES_REQUESTED** [review]
Looks good, thanks for doing this! We should've done this a long time ago probably lol...

File: engine/src/main/battlecode/server/Server.java (hunk 1 above)
Comment (stefangimmillaro) [inline engine/src/main/battlecode/server/Server.java:230]:
```suggestion
        if (archonCount < GameConstants.MIN_STARTING_ARCHONS * 2) {
```

File: engine/src/main/battlecode/server/Server.java [hunk 3]
```diff
@@ -194,7 +196,44 @@ public void run() {
...
+        
+        // Check starting Archons
+        int archonCount = 0;
+        for (RobotInfo robotInfo : liveMap.getInitialBodies()) {
+            if (robotInfo.type == RobotType.ARCHON) archonCount++;
+        }
+        if (archonCount < GameConstants.MIN_STARTING_ARCHONS * 2) {
+            throw new RuntimeException("RUBBLE BENEATH GameConstants.MIN_STARTING_ARCHONS");
+        }
+        if (archonCount > GameConstants.MAX_STARTING_ARCHONS*2) {
```
Comment (stefangimmillaro) [inline engine/src/main/battlecode/server/Server.java:233]:
```suggestion
        if (archonCount > GameConstants.MAX_STARTING_ARCHONS * 2) {
```

**Comment by stefangimmillaro** [comment]
@wflms20110333 done!

**Comment by stefangimmillaro** [comment]
@wflms20110333 
image
image

In case you didn't know about this feature, you can make suggestions in PRs. That way I can one-click merge them. I use this feature in my workplace, it improves PR resolution time. Thanks again.

**Comment by wflms20110333** [comment]
Oh wow I did not know about this feature. Seems cool, will keep in mind for the future!

**Pull request merged**